### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by dlyr**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

Be aware that the PR request cannot be accepted if it doesn't pass the Continuous Integration tests.

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
small bug/feature fix

* **What is the current behavior?** (You can also link to an open issue here)

- configure radium lib includes current source dir, which makes includes not consistent accross different files, and are prone to name clash.
- one badly pre declared class
- some pass by const values instead of ref

* **What is the new behavior (if this is a feature change)?**

fix theses

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

yes for the includes, may break client code if the do not use full include path (e.g. include Engine or Core)

* **Other information**:

### CODE REVIEW

File: src/IO/AssimpLoader/AssimpGeometryDataLoader.cpp [hunk 1]
```diff
@@ -174,7 +174,7 @@ void AssimpGeometryDataLoader::fetchVertices( const aiMesh& mesh, GeometryData&
     const int size = mesh.mNumVertices;
     auto& vertex   = data.getVertices();
     vertex.resize( mesh.mNumVertices );
-#pragma omp parallel for
+    //#pragma omp parallel for
```
Comment (MathiasPaulin) [inline src/IO/AssimpLoader/AssimpGeometryDataLoader.cpp:177]: why commenting out this pragma ?

Comment (dlyr) [reply src/IO/AssimpLoader/AssimpGeometryDataLoader.cpp:177]: should not, I revert this one (I have commented to test if it removes omp errors with valgrind)

**Comment by dlyr** [comment]
For ExamplePluginWithLibs, I'm not sure it's the right way to handle it, but in my first/quick tests I do not get how to have full include path.

**Review by MathiasPaulin — APPROVED** [review]

**Review by nmellado — COMMENTED** [review]
Fine for me, except a minor typo

File: tests/ExamplePluginWithLib/Upstream/Library/CMakeLists.txt [hunk 2]
```diff
@@ -45,6 +45,19 @@ add_library(
 
 target_link_libraries(${PROJECT_NAME} PUBLIC Radium::Core Radium::Engine)
 
+# Client libraries source code organisation is not restricted to the Radium internal configuration.
+# If the directory structure is similar to the Radium directory structure, all is managed
+# automatically by the function configure_radium_library If the directory structure is different,
+# client libraries should manage their include directories property according to the directory
```
[the remark this answers is not in the document]
Comment (dlyr) [reply tests/ExamplePluginWithLib/Upstream/Library/CMakeLists.txt:51]: I think in this case it's a cmake target property, maybe add the name of the property variable (e.g. should manage their INCLUDE_DIRECTORIES property)

[the remark this answers is not in the document]
Comment (MathiasPaulin) [reply tests/ExamplePluginWithLib/Upstream/Library/CMakeLists.txt:51]: I agree with @dlyr

[the remark this answers is not in the document]
Comment (dlyr) [reply tests/ExamplePluginWithLib/Upstream/Library/CMakeLists.txt:51]: @MathiasPaulin could you rephrase to show the property name ? (I just guess what it can be, but do not check what it is actually)

File: tests/ExamplePluginWithLib/Upstream/Library/CMakeLists.txt [hunk 3]
```diff
@@ -45,6 +45,19 @@ add_library(
 
 target_link_libraries(${PROJECT_NAME} PUBLIC Radium::Core Radium::Engine)
 
+# Client libraries source code organisation is not restricted to the Radium internal configuration.
+# If the directory structure is similar to the Radium directory structure, all is managed
+# automatically by the function configure_radium_library If the directory structure is different,
```
Comment (MathiasPaulin) [inline tests/ExamplePluginWithLib/Upstream/Library/CMakeLists.txt:50]:
```suggestion
# automatically by the function configure_radium_library. If the directory structure is different,
```

File: tests/ExamplePluginWithLib/Upstream/Library/CMakeLists.txt (hunk 2 above)
Comment (MathiasPaulin) [inline tests/ExamplePluginWithLib/Upstream/Library/CMakeLists.txt:51]:
```suggestion
# client libraries should manage their INCLUDE_DIRECTORIES property according to the directory
```

**Review by nmellado — APPROVED** [review]
